service/controller/chartconfig/v7/key/key.go

```diff
@@ -23,6 +25,14 @@ func ConfigMapNamespace(customObject v1alpha1.ChartConfig) string {
 	return customObject.Spec.Chart.ConfigMap.Namespace
 }
 
+func CordonReason(customObject v1alpha1.ChartConfig) string {
```

Comment:
Do we have to expose those functions? I.e. can't they be private?

Reply:
To not pollute the interface which leads to less maintenance effort. If funcs are private and not used in the package then they can be deleted right away w/o any cross checking.

Reply:
OK I get that. But the key package has a specific purpose and the functions are used in the chart resource. As the reason in the included in the logs.

So I think they need to be public. Unless I'm missing something?

Reply:
Yeah but those two, i.e. `CordonReason` and `CordonUntil` are used only internally so where could they be logged?

I'm not talking about `IsCordoned` to be clear. Maybe I wasn't in the first place.

Reply:
Both are logged so there is a reason why the resource is cancelled.

https://github.com/giantswarm/chart-operator/blob/a0f5065841a4c71044083a04b92b8535c515d4d4/service/controller/chart/v1/resource/release/current.go#L21

Reply:
I missed that sorry 😅  Thanks for bearing with me <3

Reply:
No problem! :)